snuba/clusters/cluster.py

```diff
@@ -0,0 +1,102 @@
+from abc import ABC, abstractmethod
+from typing import Generic, Set
+
+from snuba import settings
+from snuba.clickhouse.native import ClickhousePool, NativeDriverReader
+from snuba.clickhouse.query import ClickhouseQuery
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.reader import Reader, TQuery
+
+
+class Cluster(ABC, Generic[TQuery]):
+    """
+    A cluster is responsible for managing a collection of database nodes.
+
+    Clusters are configurable, and will be instantiated based on user defined settings.
+
+    Each storage must be mapped to a cluster via a storage set, which defines
+    the storages that must be located on the same cluster.
+
+    In future, clusters will also be responsible for co-ordinating commands that
+    need to be run on multiple hosts that are colocated within the same cluster.
+    The cluster will expose methods for:
+        - bootstrap
+        - migrate
+        - cleanup
+        - optimize
+    """
+
+    def __init__(self, storage_sets: Set[str]):
+        self.__storage_sets = storage_sets
+
+    def get_storage_sets(self) -> Set[StorageSetKey]:
+        return {StorageSetKey(storage_set) for storage_set in self.__storage_sets}
```

Comment:
For consistency with: https://github.com/getsentry/snuba/pull/860/files#r413246184

```suggestion
    def get_storage_set_keys(self) -> Set[StorageSetKey]:
        return {StorageSetKey(storage_set) for storage_set in self.__storage_sets}
```

Reply:
Thank you